homeassistant/components/suez_water/coordinator.py

```diff
@@ -48,61 +34,38 @@ def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         )
 
     async def _async_setup(self) -> None:
-        self._sync_client = await self.hass.async_add_executor_job(self._get_client)
+        self._async_client = await self._get_client()
+        self._data_api = SuezData(self._async_client)
 
     async def _async_update_data(self) -> AggregatedSensorData:
         """Fetch data from API endpoint."""
         async with asyncio.timeout(30):
-            return await self.hass.async_add_executor_job(self._fetch_data)
+            try:
+                data = await self._fetch_data()
+                _LOGGER.debug("Successfully fetched suez data")
```

Comment:
only have stuff in the try block that can raise

Reply:
Hehe, I learnt this here, but I also find the reason behind it nice, so if I ever start to program other languages again, I will definetly watch out for this because clean code